package/MDAnalysis/analysis/rms.py

```diff
@@ -535,29 +554,79 @@ def __init__(self, atomgroup, reference=None, select='all',
+        if str(self.weights) == 'mass':         # apply 'mass' weights for all selections
+           self.weights = ['mass'] * (len(self.groupselections) + 1)
+        elif self.weights is None:         # apply 'None' weights for all selections
+           self.weights = [None] * (len(self.groupselections) + 1)  
+        elif (np.array(self.weights).ndim == 1) and (np.array(self.weights).dtype 
+                                             in (np.dtype('float64'),np.dtype('int64'))):
+           none_selection = [None] * len(self.groupselections)       # apply '1D array' weights to select
+           none_selection.insert(0, self.weights)                   # & apply 'None' weights to groupselections
+           self.weights = none_selection
+
+
+
+        # add the array of weights to weights_select
+        if str(self.weights[0]) == 'mass':
+            self.weights_select = self.mobile_atoms.masses
+            self.weights_ref = self.ref_atoms.masses
+        elif self.weights[0] is None:
+            self.weights_select = None
+            self.weights_ref = None
+        else:
+            self.weights_select = self.weights[0]
+            self.weights_ref = self.weights[0]
+        if self.weights_select is not None:
+            self.weights_select = np.asarray(self.weights_select, dtype=np.float64) /  \
+                                             np.mean(self.weights_select)
+            self.weights_ref = np.asarray(self.weights_ref, dtype=np.float64) /  \
+                                             np.mean(self.weights_ref)
+        # add the array of weights to weight_groupselection
+        if self._groupselections_atoms:                            
+            for igroup, atoms in enumerate(
+                        self._groupselections_atoms, 3):
+                if str(self.weights[igroup-2]) == 'mass':
+                    self.weights_groupselection.append(atoms['mobile'].masses)
+                elif iterable(self.weights[igroup-2]):
+                    self.weights_groupselection.append(self.weights[igroup-2])
+                    self.weights_groupselection[igroup-3] =   \
+                                       np.asarray(self.weights_groupselection[igroup-3], 
+                                                                        dtype=np.float64) / \
+                                       np.mean(self.weights_groupselection[igroup-3])                  
+                else:
+                    self.weights_groupselection.append(None)
+
+                if self.weights_groupselection[igroup-3] is not None:
+                    self.weights_groupselection[igroup-3] =   \
+                                       np.asarray(self.weights_groupselection[igroup-3], 
+                                                                        dtype=np.float64) / \
+                                       np.mean(self.weights_groupselection[igroup-3])
```

Comment:
Weights set manually will end up being rescaled by the mean twice - the first rescaling is unnecessary